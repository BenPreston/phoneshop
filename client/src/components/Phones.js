import React from 'react';
import Phone from './Phone';

function Phones({phones, openPopup}) {

  return (
    <section className="results">
      <h2>Phones</h2>
        <div className="phones">
          {phones.map((phone, i ) => 
                  <Phone key={phone.id} phone={phone} openPopup={openPopup}/> 
            )}
      </div>
    </section> 
  )
}

export default Phones
