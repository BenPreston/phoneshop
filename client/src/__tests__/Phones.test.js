import React from 'react'
import Phones from '../components/Phones';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';


const fakePhones = [
    {
        id: 0,
        name: 'Galaxy S8'
    },
    {
        id: 1,
        name: 'iPhone 7'
        
    }
];

const mockOpenPopup = jest.fn();

describe('<Phone />', () => {

    it('Renders and displays properly', () => {
        const wrapper = shallow(<Phones phones={fakePhones} openPopup={mockOpenPopup} />);

        expect(toJSON(wrapper)).toMatchSnapshot();

        console.log(wrapper.debug())

    })


})