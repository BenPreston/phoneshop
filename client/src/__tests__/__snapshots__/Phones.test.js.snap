// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Phone /> Renders and displays properly 1`] = `
<section
  className="results"
>
  <h2>
    Phones
  </h2>
  <div
    className="phones"
  >
    <Phone
      key="0"
      openPopup={[Function]}
      phone={
        Object {
          "id": 0,
          "name": "Galaxy S8",
        }
      }
    />
    <Phone
      key="1"
      openPopup={[Function]}
      phone={
        Object {
          "id": 1,
          "name": "iPhone 7",
        }
      }
    />
  </div>
</section>
`;
